ui/src/app/applications/components/applications-list/applications-list.tsx

```diff
@@ -162,14 +162,26 @@ const ViewPref = ({children}: {children: (pref: AppsListPreferences & {page: num
 function filterApps(applications: models.Application[], pref: AppsListPreferences, search: string): {filteredApps: models.Application[]; filterResults: FilteredApp[]} {
     applications = applications.map(app => {
         let isAppOfAppsPattern = false;
+        const childApps = [];
         for (const resource of app.status.resources) {
             if (resource.kind === 'Application') {
                 isAppOfAppsPattern = true;
-                break;
+                childApps.push(resource.name);
             }
         }
-        return {...app, isAppOfAppsPattern};
+        return {...app, isAppOfAppsPattern, childApps};
     });
+    const parentAppArray = applications.filter(app => app.isAppOfAppsPattern);
+    if (parentAppArray.length > 0) {
+        for (const eachApp of parentAppArray) {
```

Comment:
Here we're looping through some applications for the second time (`applications.filter` is the first time).

Can we do one loop of `applications` instead? We can also use a map to avoid looping through all applications again in `findIndex` if we've already come across the child app.

```ts
const appMap: {[name: string]: number} = {}
for (const [i, app] of applications.entries()) {
  appMap[app.metadata.name] =  i;
  if (app.isAppOfAppsPattern) {
    for (const childAppName of app.childApps) {
      const indexOfChild = appMap[childAppName] || applications.findIndex(app => app.metadata.name === childAppName);
      if (indexOfChild > -1) {
        applications[indexOfChild].parentApp = app.metadata.name;
      }
    }
  }
}
```

Some users have hundreds of applications, and we already have performance issues for these cases, so it's best to avoid nested for loops whenever possible.

Reply:
Thanks for the review @rbreeze .

I tried to use the map solution and couldn't reduce the loop times. In the above solution, appMap[childAppName] won't have the right value until the first loop is done. So in order to get the appMap[childAppName] value, it has to be two different loops. Tried another way to reduce the loop times. Does it look better to you? Let me know if I misunderstood you or any feedback.